suite/suite.go

```diff
@@ -63,6 +63,20 @@ func failOnPanic(t *testing.T) {
 	}
 }
 
+// Run provides suite functionality around golang subtests.  It should be
+// called in place of t.Run(name, func(t *testing.T)) in test suite code
+// to expose setup and teardown functionality for each subtest.  The passed-in
+// func will be executed as a subtest with a fresh instance of t.  Provides
+// compatibility with go test pkg -run TestSuite/TestName/SubTestName.
+func (suite *Suite) Run(name string, subtest func()) {
+	oldT := suite.T()
+	oldT.Run(name, func(t *testing.T) {
+		suite.SetT(t)
+		defer suite.SetT(oldT)
```

Comment:
Should this be moved to before the `oldT.Run` call and outside the inner function here?

My primary concern is `oldT.Run` returning prior to the subtest finishing. According to [T.Run](https://golang.org/pkg/testing/#T.Run), this can occur if `t.Parallel` is called. If two subtests are running in parallel on this suite, that may lead to `oldT := suite.T()` in the second subtest as setting `oldT` as the testing context of the first subtest (since it is still executing  and thus does not had its defer called). Then when the second subtest finishes executing, it will set the suite testing context to the first subtest testing context instead of to the parent context as expected.

Reply:
:+1: